File: Java/src/Excel Sheet Column Title.java
import java.util.HashMap;

public class Solution
{
  public String convertToTitle(int n)
  {
    HashMap<Integer, String> map = new HashMap()
    {
      {
        put(1, "A");
        put(2, "B");
        put(3, "C");
        put(4, "D");
        put(5, "E");
        put(6, "F");
        put(7, "G");
        put(8, "H");
        put(9, "I");
        put(10, "J");
        put(11, "K");
        put(12, "L");
        put(13, "M");
        put(14, "N");
        put(15, "O");
        put(16, "P");
        put(17, "Q");
        put(18, "R");
        put(19, "S");
        put(20, "T");
        put(21, "U");
        put(22, "V");
        put(23, "W");
        put(24, "X");
        put(25, "Y");
        put(0, "Z");
      }
    };

    StringBuilder ret = new StringBuilder();
    int i = n / 26;
    int j = n % 26;
    if (i == 0)
    {
      return map.get(j);
    }

    while (i > 0)
    {
      i = n / 26;
      j = n % 26;

      if (j == 0)
      {
        i--;
      }
      ret.insert(0, map.get(j));
      n = i;
    }
    // ret.reverse();
    return ret.toString();
  }
}

File: Python/Excel Sheet Column Title.py
class Solution:
    # @return a string
    def convertToTitle(self, num):
               
        map = {1:'A', 2:'B',3:'C',4:'D',5:'E',6:'F',7:'G',8:'H',9:'I',10:'J',11:'K',12:'L',13:'M',14:'N',15:'O',16:'P', 17:'Q',18:'R',19:'S',20:'T',21:'U',22:'V',23:'W', 24:'X',25:'Y',0:'Z'}
        
        ret = []
        i,j = divmod(num,26)
 
        if i==0:
            return map[j]
        while i > 0:
            i,j = divmod(num,26)
            if j==0:
                i = i-1
            ret.append(map[j])
            num = i 
        ret.reverse()
        return "".join(ret)